content/test/unit/service/deployments/deployments.spec.ts

```diff
@@ -0,0 +1,151 @@
+import { Deployment, EntityType, EntityVersion, PartialDeploymentHistory } from 'dcl-catalyst-commons'
+import { restore, stub } from 'sinon'
+import { AppComponents } from 'src/types'
+import { ContentFilesRow } from '../../../../src/logic/database-queries/content-files-queries'
+import { HistoricalDeploymentsRow } from '../../../../src/logic/database-queries/deployments-queries'
+import { MigrationDataRow } from '../../../../src/logic/database-queries/migration-data-queries'
+import {
+  getCuratedLimit,
+  getCuratedOffset,
+  getDeployments,
+  MAX_HISTORY_LIMIT
+} from '../../../../src/service/deployments/deployments'
+import { DeploymentOptions } from '../../../../src/service/deployments/types'
+
+describe('deployments service', () => {
+  describe('getDeployments', () => {
+    let components: Pick<AppComponents, 'database'>
+    let result: PartialDeploymentHistory<Deployment>
+
+    const deploymentIds = [127, 255]
+
+    const historicalDeploymentsRows: HistoricalDeploymentsRow[] = [
+      {
+        auth_chain: [],
+        deleter_deployment: 1,
+        deployer_address: '1',
+        entity_id: '1',
+        entity_metadata: {},
+        entity_pointers: ['0,0'],
+        entity_timestamp: Date.now(),
+        entity_type: EntityType.SCENE,
+        id: deploymentIds[0],
+        local_timestamp: Date.now(),
+        version: EntityVersion.V3
+      },
+      {
+        auth_chain: [],
+        deleter_deployment: 1,
+        deployer_address: '2',
+        entity_id: '2',
+        entity_metadata: {},
+        entity_pointers: ['0,1'],
+        entity_timestamp: Date.now(),
+        entity_type: EntityType.SCENE,
+        id: deploymentIds[1],
+        local_timestamp: Date.now(),
+        version: EntityVersion.V3
+      }
+    ]
+
+    const contentFiles: ContentFilesRow[] = [
+      {
+        content_hash: 'hash',
+        deployment: deploymentIds[0],
+        key: 'key'
+      },
+      {
+        content_hash: 'hash1',
+        deployment: deploymentIds[1],
+        key: 'key1'
+      }
+    ]
+
+    const migrationData: MigrationDataRow[] = [
+      {
+        deployment: deploymentIds[0],
+        original_metadata: []
+      },
+      {
+        deployment: deploymentIds[1],
+        original_metadata: []
+      }
+    ]
+
+    const options: DeploymentOptions = {
+      filters: {
+        from: Date.now(),
+        to: Date.now()
+      }
+    }
+
+    beforeAll(async () => {
+      components = { database: { queryWithValues: () => {} } as any }
+      stub(components.database, 'queryWithValues')
+        .onFirstCall()
+        .resolves({ rows: historicalDeploymentsRows, rowCount: 2 })
+        .onSecondCall()
+        .resolves({ rows: contentFiles, rowCount: 2 })
+        .onThirdCall()
+        .resolves({ rows: migrationData, rowCount: 2 })
+
+      result = await getDeployments(components, options)
+
+      console.log('RESULT', result)
+    })
```

Comment:
When awaiting funcitons prefer `it` instead of `beforeAll`. If `beforeAll` fails then the error goes to /dev/null and testing becomes more difficult